### PR TITLE
Enable `bf16` support in the benchmark scripts

### PR DESCRIPTION
**Pull request by yanbing-j**
This PR is to enable bf16 support for benchmark. Currently, kernel, citation, points and inference benchmark scripts support bf16.

### CODE REVIEW

**Comment by yanbing-j** [comment]
> Looks great! Thank you.
Thanks! I'm adding bf16 support of kernel/points/inference of benchmark. Will update this PR soon.

**Comment by yanbing-j** [comment]
Hi @rusty1s , I have updated this PR. Please help review again. Thanks!

File: benchmark/citation/train_eval.py [hunk 1]
```diff
@@ -90,34 +90,40 @@ def run_train(dataset, model, runs, epochs, lr, weight_decay, early_stopping,
...
+def run_inference(dataset, model, epochs, profiling, bf16, permute_masks=None,
                   logger=None):
     data = dataset[0]
     if permute_masks is not None:
         data = permute_masks(data, dataset.num_classes)
     data = data.to(device)
 
     model.to(device).reset_parameters()
+    amp = torch.cuda.amp.autocast(enabled=False) if torch.cuda.is_available(
+    ) else torch.cpu.amp.autocast(enabled=bf16)
```
Comment (rusty1s) [inline benchmark/citation/train_eval.py:102]:
```suggestion
    if torch.cuda.is_available():
            amp = torch.cuda.amp.autocast(enabled=False)
        else:
            amp = torch.cpu.amp.autocast(enabled=args.bf16)
```

File: benchmark/kernel/main_performance.py [hunk 2]
```diff
@@ -44,6 +45,9 @@
 ]
 
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+amp = torch.cuda.amp.autocast(
+    enabled=False) if torch.cuda.is_available() else torch.cpu.amp.autocast(
+        enabled=args.bf16)
```
Comment (rusty1s) [inline benchmark/kernel/main_performance.py:50]:
```suggestion
if torch.cuda.is_available():
    amp = torch.cuda.amp.autocast(enabled=False)
else:
    amp = torch.cpu.amp.autocast(enabled=args.bf16)
```

File: benchmark/points/train_eval.py [hunk 3]
```diff
@@ -41,31 +41,34 @@ def run_train(train_dataset, test_dataset, model, epochs, batch_size, lr,
 
 
 @torch.no_grad()
-def run_inference(test_dataset, model, epochs, batch_size, profiling):
+def run_inference(test_dataset, model, epochs, batch_size, profiling, bf16):
     model = model.to(device)
     test_loader = DataLoader(test_dataset, batch_size, shuffle=False)
+    amp = torch.cuda.amp.autocast(enabled=False) if torch.cuda.is_available(
+    ) else torch.cpu.amp.autocast(enabled=bf16)
```
Comment (rusty1s) [inline benchmark/points/train_eval.py:48]:
```suggestion
    if torch.cuda.is_available():
        amp = torch.cuda.amp.autocast(enabled=False)
    else:
        amp = torch.cpu.amp.autocast(enabled=args.bf16)
```

**Review by rusty1s — APPROVED** [review]

**Pull request merged**